keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -221,6 +222,16 @@ public void addParent(Vertex<?> parent) {
         return parents;
     }
 
+    public Vertex<T> print() {
```

Comment:
If you do the following you can avoid overriding on the subclasses.
```
public <V extends Vertex<T>> V print() {
        new PrintVertex<>(this);
        return (V) this;
    }
```

Reply:
Ah great, have removed implementation from subclasses